### PR TITLE
Bumping version to 2.0.1 for opensearch-transport

### LINKED ISSUE #91: Release opensearch-transport updates with Faraday 2 support
### Is your feature request related to a problem?
#59 is fixed, but not released.

### What solution would you like?
Release opensearch-transport 1.0.1 and 2.0.1.

### Do you have any additional context?
Can the opensearch gem's version restriction for opensearch-transport be changed from `"2.0.0"` and `"1.0.0"`, to `"~> 2.0.0"` and `"~> 1.0.0"`, respectively? It probably doesn't need to be more specific, but if it does, then change it to `"2.0.1"` and `"1.0.1"` respectively.

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Bumping version to 2.0.1 for opensearch-transport.

### Issues Resolved
#91 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VachaShah** [comment]
@BrianHawley I am not able to add you for reviewer, so tagging you here.

**Review by dblock — APPROVED** [review]

File: opensearch/opensearch.gemspec [hunk 1]
```diff
@@ -61,7 +61,7 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>= 2.4'
 
-  s.add_dependency 'opensearch-transport', '2.0.0'
+  s.add_dependency 'opensearch-transport', '~> 2.0.0'
```
Comment (saratvemulapalli) [inline opensearch/opensearch.gemspec:64]: Curious why does the client rely on `opensearch-transport`?

Comment (VachaShah) [reply opensearch/opensearch.gemspec:64]: So the ruby client has 3 gems which are sub-gems kind of, `opensearch-api`, `opensearch-transport` and `opensearch-dsl`. They are all written in the client and created as separate gems plus they are used in the main gem which is the `opensearch-ruby` gem. So all the 3 gems can be used separately or can be used as part of the `opensearch-ruby` gem.

**Comment by BrianHawley** [comment]
This is also going to need a release of the opensearch gem in order to enable that dependency restriction change.

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**